### PR TITLE
Docs/redux

### PR DESCRIPTION
**Pull request by jevakallio**

### CODE REVIEW

File: docs/SETUP.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    $ npm install
+    $ cp env.example.js env.js
+
+### Configure Auth0
+
+1. Before you start you need to create a new application in [Auth0](https://manage.auth0.com/#/applications/)
+2. Set `AUTH0_CLIENT_ID` and `AUTH0_NAMESPACE` in `env.js` according to your application you created in Auth
+
+       AUTH0_CLIENT_ID: '<CLIENT_ID>',
+       AUTH0_NAMESPACE: '<APP_NAME>.eu.auth0.com'
```
Comment (tino-junge) [inline docs/SETUP.md]: Formatting looks off here

File: docs/ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+The application state and state changes are managed by **Redux**, a library that implements a pure, side-effect-free variant of the Facebook Flux architecture. Redux and Flux prescribe a unidirectional dataflow through your application. To understand Redux, check out this [Cartoon guide by Lin Clark](https://code-cartoons.com/a-cartoon-guide-to-flux-6157355ab207#.4dpmozm9v) (it's great, not a joke!) and [Dan Abramov's Redux course on egghead.io](https://egghead.io/series/getting-started-with-redux).
+
+Redux helps us with synchronous updating of our state, but it doesn't provide an out-of-the-box solution for handling asynchronous actions. The Redux ecosystem has many possible solutions for this problem. In our application we use the vanilla redux-thunk middleware for simple asynchronous actions, and **redux-loop** to handle more complex asynchronisity.
+
+The state in Redux applications should never be mutated, but always cloned. To make this more natural for the programmer, and more-fault tolerant against accidental mutation, we use **ImmutableJS** data structures to hold our app's state.
+
+## Organising code
+
+Let's take a look of how we organise our application.
```
Comment (tino-junge) [inline docs/ARCHITECTURE.md]: we could add a directory structure here for a general overview (or link to our main folder in the code)

**Comment by tino-junge** [comment]
Very detailed architecture description. LGTM 👍

**Comment by jevakallio** [comment]
TODO

 * [ ] Add directory structure overview
 * [ ] More introductory stuff in README.md
 * [ ] Describe missing parts (redux, services, utils)
 * [ ] Document testing practices

**Pull request merged**